appscale/tools/appscale.py

```diff
@@ -403,25 +403,30 @@ def ssh(self, node):
     contents = self.read_appscalefile()
     contents_as_yaml = yaml.safe_load(contents)
 
-    # make sure the user gave us an int for node
-    try:
-      index = int(node)
-    except ValueError:
-      raise TypeError("Usage: appscale ssh <node id to ssh to>")
-
     if 'keyname' in contents_as_yaml:
       keyname = contents_as_yaml['keyname']
     else:
       keyname = "appscale"
 
-    nodes = self.get_nodes(keyname)
-    # make sure there is a node at position 'index'
+    if node is None:
+      node = "shadow"
+
     try:
+      index = int(node)
+      nodes = self.get_nodes(keyname)
+      # make sure there is a node at position 'index'
       ip = nodes[index]['public_ip']
     except IndexError:
       raise AppScaleException("Cannot ssh to node at index " +
-        str(index) + ", as there are only " + str(len(nodes)) +
-        " in the currently running AppScale deployment.")
+                              ", as there are only " + str(len(nodes)) +
```

Comment:
@tmarballi we would get "ssh: Could not resolve hostname : Name or service not known". I think most people would be just using "appscale ssh" to get to the head node... maybe I could put a help text that would explain like: "use an index or a role in [list of roles], if you have a deployment with private ips this may not work". However currently the appscale ssh [index] doesn't work if there are private ips.